.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,36 @@
+name: Pack and publish nugets
+
+on:
+  release:
+    types:
+      - published
+
+jobs:
+  build-pack:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install dotnet6
+        uses: actions/setup-dotnet@v2
+        with:
+          dotnet-version: |
+            5.0.x
+            6.0.x
+      - name: Install deps
+        run: |
+          dotnet restore
+      - name: Build
+        run: |
+          dotnet build --configuration Release --no-restore -p:Deterministic=true -p:BuildNumber=${{ github.run_number }}
```

Comment:
Do we need both build and pack? Won't pack also build? I see you have specified --no-build on pack, or do you want to know if the build fails specifically?

Reply:
Based the implementation on https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/.github/workflows/publish-packages-1.0.yml 
The idea is to more clearly display what part of the pack process (restore, build, pack) that fails.